Loop/View Controllers/CarbAbsorptionViewController.swift

```diff
@@ -121,15 +121,17 @@ final class CarbAbsorptionViewController: LoopChartsTableViewController, Identif
         let size = currentContext.newSize ?? self.tableView.bounds.size
         let availableWidth = size.width - self.charts.fixedHorizontalMargin
         let totalHours = floor(Double(availableWidth / minimumSegmentWidth))
+        let pastHours = totalHours - 1 // keep 1 hour for presenting current data and looking into the future
```

Comment:
So it took me a while to track down and remember how this works. The end date of the graph is set by the last counteraction effects, plus one hour, rounded in a future direction to the next even hour. So when we do have data, the chart typically shows 1-2 hours of future data. The changes here mostly affect when we do not have data, and seem fine (it doesn't matter all that much what we show for axis when there is no data displayed).  But the one effect that it does have when we do have data is that it is shortening the historical view by one hour. 

I don't think that is intentional. To fix the no-data case, I'd recommend instead that we remove the "pastHours" var, and just bump the end date instead.

Reply:
I'll put an example below.